packages/runtime/runtime-utils/src/requestParser.ts

```diff
@@ -30,6 +30,13 @@ export class RequestParser implements IRequest {
     private requestPathParts: readonly string[] | undefined;
     public readonly query: string;
     constructor(private readonly request: Readonly<IRequest>) {
+        // Perf optimizations. Even better would be not to create this object
```

Comment:
looks like `new RequestParser` is only called in a few places; if this is worth optimizing here, would it also be worth just adding a create method that does this check but spits back the same request under this condition?